src/MongoDB.Driver.Core/Core/Operations/FindOneAndDeleteOperation.cs

```diff
@@ -116,6 +137,7 @@ internal override BsonDocument CreateCommand(ICoreSessionHandle session, Connect
                 { "maxTimeMS", () => MaxTimeHelper.ToMaxTimeMS(_maxTime.Value), _maxTime.HasValue },
                 { "writeConcern", writeConcern, writeConcern != null },
                 { "collation", () => Collation.ToBsonDocument(), Collation != null },
+                { "hint", () => _hint, _hint != null },
```

Comment:
You could actually simplify this a bit to:

    { "hint", _hint, _hint != null },

because `_hint` does no work and has no side effects.

Reply:
Done.